server/src/middlewares/userValidator.js

```diff
@@ -13,43 +13,19 @@ export default class userValidator {
    * @returns {Object} Success or failure message
    */
   static signUp(req, res, next) {
-    const keys = Object.keys(req.body);
-    let flag = false;
-    keys.forEach((key) => {
-      if (!isString(req.body[key])) {
-        flag = true;
-        return res.status(400).json({ message: `Invalid Format for ${key} field` });
-      }
-    });
-    if (flag) return;
+    if (req.body.firstName && req.body.firstName.length > 20) return res.status(400).json({ message: 'First name cannot be more than 20 characters' });
+    if (req.body.lastName && req.body.lastName.length > 20) return res.status(400).json({ message: 'Last name cannot be more than 20 characters' });
+    if (req.body.email && req.body.email.length > 30) return res.status(400).json({ message: 'Email cannot be more than 30 characters' });
+    if (req.body.password && req.body.password.length > 20) return res.status(400).json({ message: 'password cannot be more than 20 characters' });
+    if (isStringValidator(req, res) === null) return;
     if (!req.body.firstName || !req.body.lastName || !req.body.email || !req.body.password) {
       return res.status(400).json({ message: 'All fields are required' });
     }
     if (req.body.email.indexOf('@') === -1 || req.body.email.indexOf('.') === -1) {
       return res.status(400).json({ message: 'Invalid email address' });
     }
-    if (req.body.firstName && req.body.firstName.indexOf('-') !== -1) {
-      let test = req.body.firstName;
-      for (let i = 0; i < test.length; i += 1) {
-        test = test.replace('-', '');
-      }
-      if (!validator.isAlpha(test)) {
-        return res.status(400).json({ message: 'First name can only contain alphabelts and hyphen' });
-      }
-    } else if (!validator.isAlpha(req.body.firstName)) {
-      return res.status(400).json({ message: 'First name can only contain alphabelts and hyphen' });
-    }
-    if (req.body.lastName && req.body.lastName.indexOf('-') !== -1) {
-      let test = req.body.lastName;
-      for (let i = 0; i < test.length; i += 1) {
-        test = test.replace('-', '');
-      }
-      if (!validator.isAlpha(test)) {
-        return res.status(400).json({ message: 'Last name can only contain alphabelts and hyphen' });
-      }
-    } else if (!validator.isAlpha(req.body.lastName)) {
-      return res.status(400).json({ message: 'Last name can only contain alphabelts and hyphen' });
-    }
+    if (checkName(req.body.firstName) === null) return res.status(400).json({ message: 'First name can only contain alphabelts and hyphen' });
```

Comment:
I don't understand this check,``` firstname === null``` but the response is ``` 'First name can only contain alphabelts and hyphen' ```. Can't length be used for this test ?

Reply:
Checkname is a function that verifies if the string passed contains only alphabets and hyphen. I am calling that function there and passing in req.body.firstname as name parameter. If the test is negative it returns null, that's what I was trying to do. what do you think please ?

Reply:
I understand but you are also checking if the firstname is equal to null. I think this can be covered using the length property of your validators